### PR TITLE
DB-10961 Get rid of the NodeFactory part 3

### PR DESCRIPTION
**Pull request by martinrupp**
iterative PR to get rid of NodeFactory / getNode

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3011/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3084/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3086/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Review by arnaud-splice — APPROVED** [review]
Awesome!

**Review by dgomezferro — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/StatementNode.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
 import com.splicemachine.db.iapi.services.loader.GeneratedClass;
 import com.splicemachine.db.iapi.services.sanity.SanityManager;
 import com.splicemachine.db.iapi.sql.ResultDescription;
+import com.splicemachine.db.iapi.sql.compile.C_NodeTypes;
```
[the remark this answers is not in the document]
Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/StatementNode.java:43]: fixed in part 4

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/UntypedNullConstantNode.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
 import com.splicemachine.db.iapi.services.context.ContextManager;
 import com.splicemachine.db.iapi.services.sanity.SanityManager;
 import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.sql.compile.C_NodeTypes;
```
[the remark this answers is not in the document]
Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/UntypedNullConstantNode.java:38]: fixed in part 4

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/3134/

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**